src/pollen.py

```diff
@@ -0,0 +1,63 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Pollen charm business logic."""
+
+import os
+import shutil
+import time
+from glob import glob
+from subprocess import run
+
+from charms.operator_libs_linux.v0 import apt
+from charms.operator_libs_linux.v1 import systemd
+
+from exceptions import ConfigurationWriteError, InstallError
+
+
+def prepare() -> None:
+    """Install packages and write configuration files.
+
+    Raises:
+        InstallError: if the packages fail to install
+        ConfigurationWriteError: something went wrong writing the configuration
+    """
+    try:
+        apt.update()
+        apt.add_package(["pollen", "pollinate", "ent"])
+        # Apt-get makes the pollen restart error if we don't wait,
+        # even if apt-get executes correctly and there are no
+        # child processes left. The cause is unknown.
+        time.sleep(10)
```

Comment:
Why this `sleep`?

Reply:
for some unknown reason, the charm errors when trying to restart the pollen service for the first time and fixes itself in the second attempt.I have tried to track & isolate the error but it seems that the installation of the pollen package triggers processes after apt-get finished its execution that I cannot monitor in the code, so I decided to just wait in order to fix that for now.

Reply:
This seems like something that should be tracked down, what if the 10 second sleep is not enough in some cases? I guess the charm will error in that case?

Reply:
I tried to track down the source of the failure, but even though it happens after the apt-get install command, it is not associated with the apt-get install process termination.